nni/tools/nnictl/config_utils.py

```diff
@@ -54,39 +56,53 @@ class Experiments:
     def __init__(self, home_dir=NNICTL_HOME_DIR):
         os.makedirs(home_dir, exist_ok=True)
         self.experiment_file = os.path.join(home_dir, '.experiment')
-        self.experiments = self.read_file()
+        self.lock = get_file_lock(self.experiment_file, timeout=1, stale=2)
```

Comment:
what is the problem that we try to solve with `timeout=1, stale=2` ?

Reply:
the name `timeout` parameter of  `get_file_lock` and `SimpleFileLock` is a little bit misleading, it is used in an inner loop, not the overall timeout.

Reply:
We will check if the lock file is modified over two seconds to determine whether the current lock need to be forced released. 

This because we assume that all operations on the `.experiment` file should be completed within `2` seconds. If the lock file has not been modified for more than `2` seconds, we think the process that generated the lock may have crashed without releasing the lock. 

The `async lock` implementation in TS code has the same logic. The only difference is that TS only retries to lock 100 times, with an interval of 0.1s, then it will throw error. In python code lock will non-stop retry and print warning with an interval of `timeout` because we think users has the ability to judge when `nnictl` should be forced to end.

`timeout=1` means we will check if lockfile is expired with an interval of `1` second.

Reply:
Indeed, `timeout` -> `check_interval`, `SimpleFileLock` -> `ExpiredFileLock` maybe better?

Reply:
Got it, then there should be error or at least warning message in the log while removing `self._lock_file`

Reply:
We could solve this problem  with special handling the timeout exception, no need to introduce `stale`
`check_interval` is good,  `SimpleFileLock` -> `SimplePreemptiveLock`
```
def __enter__()
    while True:
        try:
            self.aquire()
            return self
        except Timeout:
            # here remove self._lock_file
```